tun/tun_illumos.go

```diff
@@ -0,0 +1,271 @@
+/* SPDX-License-Identifier: MIT
+ *
+ * Copyright 2019 Joyent, Inc.
+ */
+
+// +build illumos,amd64
+
+package tun
+
+import (
+	"fmt"
+
+	"golang.org/x/sys/unix"
+	"os"
+)
+
+// NativeTun is the OS-specific implementation of a tun interface.
+type NativeTun struct {
+	tunFile *os.File /* TUN device file */
+	ip_fd   int      /* IP device fd */
+	name    string   /* Interface name */
+	mtu     int
+
+	events chan Event
```

Comment:
There's still no code to actually generate events -- MTU-change and interface up/down.